diff --git a/node_modules/react-spotify-web-playback/src/index.tsx b/node_modules/react-spotify-web-playback/src/index.tsx
index 0593ed1..5ed2f5d 100644
--- a/node_modules/react-spotify-web-playback/src/index.tsx
+++ b/node_modules/react-spotify-web-playback/src/index.tsx
@@ -48,6 +48,7 @@ import {
   STATUS,
   TYPE,
   validateURI,
+  isEqualNumber
 } from './utils';
 
 class SpotifyWebPlayer extends React.PureComponent<Props, State> {
@@ -183,11 +184,11 @@ class SpotifyWebPlayer extends React.PureComponent<Props, State> {
       ? !isEqualArray(previousProps.uris, uris)
       : previousProps.uris !== uris;
     const playOptions = this.getPlayOptions(uris);
-
+    const changedOffset = previousProps.offset !== offset
     const canPlay = !!currentDeviceId && !!(playOptions.context_uri || playOptions.uris);
-    const shouldPlay = (changedURIs && isPlaying) || !!(isReady && (autoPlay || playProp));
+    const shouldPlay = changedOffset || !!(isReady && (autoPlay || playProp));
 
-    if (canPlay && shouldPlay) {
+    if (shouldPlay) {
       await play(token, { deviceId: currentDeviceId, offset, ...playOptions });
 
       /* istanbul ignore else */
diff --git a/node_modules/react-spotify-web-playback/src/utils.ts b/node_modules/react-spotify-web-playback/src/utils.ts
index e9646e3..88f1715 100644
--- a/node_modules/react-spotify-web-playback/src/utils.ts
+++ b/node_modules/react-spotify-web-playback/src/utils.ts
@@ -51,6 +51,15 @@ export function getSpotifyURIType(uri: string): string {
   return type;
 }
 
+// export function isEqualNumber(a?: any, b?: any){
+//   console.log(a, b)
+// if (a === b){
+//   return true
+// } else if (a !== b){
+//   return false
+// }
+// }
+
 export function isEqualArray(A?: any, B?: any) {
   if (!Array.isArray(A) || !Array.isArray(B) || A.length !== B.length) {
     return false;
